impexp-client-gui/src/main/java/org/citydb/gui/operation/common/filter/XMLQueryView.java

```diff
@@ -530,5 +534,30 @@ private boolean isDefaultDatabaseSrs(DatabaseSrs srs) {
                 || (databaseController.isConnected()
                 && srs.getSrid() == databaseController.getActiveDatabaseAdapter().getConnectionMetaData().getReferenceSystem().getSrid());
     }
+
+    private boolean connectToDatabase(String message) {
```

Comment:
This method is very useful. How about move it to a util class for public use?

Reply:
Hm, what about creating a `DatabaseConnectionDialog` in _org.citydb.gui.dialog_ with a static `boolean show(message, viewController)` method. It would return true or false depending on whether the database connection could be established or not (so you would not know whether the connection was not established because the user clicked 'Cancel' or because there were other issues when trying to connect).

Reply:
sounds good.

Reply:
Ok, please check 36f5a5334c9c412e204636e02345562157164495

Reply:
Nice